### PR TITLE
Add socket activation support for server

### LINKED ISSUE #228: Add systemd socket activation support to server mode

### PR DESCRIPTION
**Pull request by eyakubovich**
Fixes #228

### CODE REVIEW

**Comment by jonboulle** [comment]
can you add a note on this to the --listen flag and a doc example?

**Comment by eyakubovich** [comment]
actually meant to but forgot. will add.

**Comment by MohdAhmad** [comment]
LGTM

File: remote/server.go [hunk 1]
```diff
@@ -153,6 +156,51 @@ func bindHandler(h handler, ctx context.Context, sm subnet.Manager) http.Handler
...
+	}
+
+	if listeners[fdOffset] == nil {
+		return nil, fmt.Errorf("fd %v was not socket activated", addr)
+	}
+
+	return listeners[fdOffset], nil
+}
+
+func listener(addr string) (net.Listener, error) {
```
Comment (jonboulle) [inline remote/server.go]: Tests for this one would be great.

**Comment by jonboulle** [comment]
LGTM although would be nice to have an example in the readme, and the tests

**Comment by eyakubovich** [comment]
The readme is getting large. We will soon break it up into Documentation/ and then can have a separate document for socket activation with example.

**Pull request merged**